backend/src/test/java/com/google/backend/servlets/AlertsDataServletTest.java

```diff
@@ -95,6 +99,41 @@ public void doGet_ReturnsAlertEntitiesAsJson() throws IOException, ServletExcept
     assertEquals(expected, result);
   }
 
+  @Test
+  public void doGet_ReturnsLimitedNumberOfSortedAlerts() throws IOException, ServletException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Alert newAlertOne = Alert.createAlertWithoutId(Timestamp.getDummyTimestamp(0), 
+        Arrays.asList(Anomaly.getDummyAnomaly()), Alert.StatusType.UNRESOLVED);
+    Alert newAlertTwo = Alert.createAlertWithoutId(Timestamp.getDummyTimestamp(1), 
+        Arrays.asList(Anomaly.getDummyAnomaly()), Alert.StatusType.UNRESOLVED);
+    Alert newAlertThree = Alert.createAlertWithoutId(Timestamp.getDummyTimestamp(2), 
+        Arrays.asList(Anomaly.getDummyAnomaly()), Alert.StatusType.UNRESOLVED);
+    datastore.put(newAlertOne.toEntity());
+    datastore.put(newAlertTwo.toEntity());
+    datastore.put(newAlertThree.toEntity());
+
+    when(request.getParameter(LIMIT_PARAM)).thenReturn(FAKE_LIMIT);
+
+
+    List<Alert> expectedAlerts = new ArrayList<>();
+    try {
+      Entity alertThree = datastore.get(KeyFactory.createKey(Alert.ALERT_ENTITY_KIND, 3));
+      Entity alertTwo = datastore.get(KeyFactory.createKey(Alert.ALERT_ENTITY_KIND, 2));    
+      expectedAlerts.add(Alert.createAlertFromEntity(alertThree));
+      expectedAlerts.add(Alert.createAlertFromEntity(alertTwo));
+    } catch (EntityNotFoundException e) {
+      throw new ServletException(e);
+    }
```

Comment:
why are you throwing the ServletException on the test?

Reply:
I changed it to a warning log (in general, the exception shouldn't ever be thrown)